src/core/i18n/utils.js

```diff
@@ -73,24 +73,93 @@ export function getLanguage(langOrLocale) {
 }
 
 export function isRtlLang(lang) {
-  const language = getLanguage(lang);
+  const language = sanitizeLanguage(lang);
   return rtlLangs.includes(language);
 }
 
 export function getDirection(lang) {
   return isRtlLang(lang) ? 'rtl' : 'ltr';
 }
 
-export function getLangFromRouter(renderProps) {
-  if (renderProps) {
-    // Get the lang from the url param by default
-    // if it exists.
-    if (renderProps.params && renderProps.params.lang) {
-      return getLanguage(renderProps.params.lang);
-    } else if (renderProps.location && renderProps.location.query &&
-               renderProps.location.query.lang) {
-      return getLanguage(renderProps.location.query.lang);
+
+function qualityCmp(a, b) {
+  if (a.quality === b.quality) {
+    return 0;
+  } else if (a.quality < b.quality) {
+    return 1;
+  }
+  return -1;
+}
+
+/*
+ * Parses the HTTP accept-language header and returns a
+ * sorted array of objects. Example object:
+ * {
+ *   lang: 'pl', quality: 0.7
+ * }
+ */
+export function parseAcceptLanguage(header) {
+  // pl,fr-FR;q=0.3,en-US;q=0.1
+  if (!header || !header.split) {
+    return [];
+  }
+  const rawLangs = header.split(',');
+  const langList = rawLangs.map((rawLang) => {
+    const parts = rawLang.split(';');
+    let q = 1;
+    if (parts.length > 1 && parts[1].indexOf('q=') === 0) {
```

Comment:
Here's an example from [Wikipedia](https://en.wikipedia.org/wiki/Content_negotiation): `Accept-Language: de; q=1.0, en; q=0.5`. I don't think this would parse it.

Maybe we use a URL parsing library to handle this? Example: https://tonicdev.com/5728c8c86ad20d1100700899/578937d636f3d6120002d359